core/modules/replica/SqlJob.cc

```diff
@@ -0,0 +1,260 @@
+/*
+ * LSST Data Management System
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+// Class header
+#include "replica/SqlJob.h"
+
+// System headers
+#include <stdexcept>
+
+// Qserv headers
+#include "replica/Configuration.h"
+#include "replica/ServiceProvider.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+using namespace std;
+
+namespace {
+
+LOG_LOGGER _log = LOG_GET("lsst.qserv.replica.SqlJob");
+
+} /// namespace
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+string SqlJob::typeName() { return "SqlJob"; }
+
+
+Job::Options const& SqlJob::defaultOptions() {
+    static Job::Options const options{
+        2,      /* priority */
+        false,  /* exclusive */
+        true    /* exclusive */
```

Comment:
exclusive is false and true? 🙃 
maybe you can elaborate on these comments?

Reply:
FWIW I *do* appreciate the comments on opaque variables. (what does true or false mean in these contexts? who knows? 😆).

Reply:
Fixed the comment. In general, I agree with your observation. In this specific case the function creates an object of the following type:
```
# file: replica/Job.h
class Job {

    /// The job options container
    struct Options {

        /// the priority level
        int  priority;

        /// the flag indicating of this job can't be run simultaneously along with other jobs.
        bool exclusive;

        /// the flag indicating if the job is allowed to be interrupted by other jobs.
        bool preemptable;
    };
    ...
```
Right now, the only meaningful parameter here is the `priority`, which affects an order in which requests are processed by the worker services. Others were reserved for a (non-existing yet) job scheduler. And I'm not sure if I want to have the one in the Replication system. That scheduler is like Buttler or SuperTask - something which would be a universal solution for everything 😄